### PR TITLE
Add canary stage to FRS stress tests runs

### PR DESCRIPTION
**Pull request by RishhiB**
Currently, the FRS stress test runs against the prod tenants and service endpoint. We want to add a new canary state to our stress test which is basically an endpoint where FRS deploys their code before it goes to production. 

Acceptance criteria:
- The new tenant secrets and service endpoint is added to the key-vault
- The FRS stress test succeeds after adding this new canary endpoint.

### CODE REVIEW

File: tools/pipelines/test-real-service-stress.yml [hunk 1]
```diff
@@ -123,3 +123,22 @@ stages:
         env:
           fluid__test__driver__frs: $(automation-fluid-test-driver-frs-stress-test)
           FLUID_TEST_LOGGER_PKG_PATH: ${{ variables.testWorkspace }}/node_modules/@ff-internal/aria-logger # Contains getTestLogger impl to inject
+  # stress tests canary
+  - stage:
+    displayName: Stress tests - canary
```
Comment (anthony-murphy) [inline tools/pipelines/test-real-service-stress.yml:128]: this should specify frs canary, and so should everything in this step

**Comment by sonalideshpandemsft** [comment]
change the name to `frs-canary` or similar to avoid confusion later. Also, did you try to run stress tests on this branch as we discussed?

**Review by sonalideshpandemsft — APPROVED** [review]

**Comment by sonalideshpandemsft** [comment]
Canary pipeline run succeeded: https://dev.azure.com/fluidframework/internal/_build/results?buildId=152245&view=logs&s=3781c43d-9993-5962-e57e-cb19c0b97d45

File: tools/pipelines/test-real-service-stress.yml [hunk 2]
```diff
@@ -123,3 +123,22 @@ stages:
...
+    - group: stress-frs-canary
+    jobs:
+    - template: templates/include-test-real-service.yml
+      parameters:
+        poolBuild: Large
+        testPackage: ${{ variables.testPackage }}
+        testWorkspace: ${{ variables.testWorkspace }}
+        artifactBuildId: $(resources.pipeline.client.runID)
+        timeoutInMinutes: 120
+        testCommand: start:frs
```
Comment (anthony-murphy) [inline tools/pipelines/test-real-service-stress.yml:141]: how is this different from the above? will the driver know it is targeting canary and set the endpoint accordingly? this is important as it is necessary differentiation in the logs and alerts

Comment (RishhiB) [reply tools/pipelines/test-real-service-stress.yml:141]: When downloading the secrets from the prague-vault, I can see it pulling from canery; https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_apis/build/builds/152245/logs/41

Comment (sonalideshpandemsft) [reply tools/pipelines/test-real-service-stress.yml:141]: How and where is the driver setting the endpoint? 

I think it is here: https://github.com/microsoft/FluidFramework/blob/main/packages/test/test-driver-definitions/src/interfaces.ts#L15 and we should add a similar endpoint as `frs-canary`.

Also, the `start:frs` script is pointing to driverendpoint as `frs`: https://github.com/microsoft/FluidFramework/blob/main/packages/test/test-service-load/package.json#L39

Comment (anthony-murphy) [reply tools/pipelines/test-real-service-stress.yml]: change look good to me now. please make sure the automation kusto data looks good too. e.g. that you can see canary runs independently from other frs runs.

File: tools/pipelines/test-real-service-stress.yml [hunk 3]
```diff
@@ -123,3 +123,22 @@ stages:
...
+    - template: templates/include-test-real-service.yml
+      parameters:
+        poolBuild: Large
+        testPackage: ${{ variables.testPackage }}
+        testWorkspace: ${{ variables.testWorkspace }}
+        artifactBuildId: $(resources.pipeline.client.runID)
+        timeoutInMinutes: 120
+        testCommand: start:frs:canary
+        env:
+          fluid__test__driver__frs: $(automation-fluid-driver-frs-canary-stress-test)
```
Comment (sonalideshpandemsft) [inline tools/pipelines/test-real-service-stress.yml:143]: we can change this env to `fluid__test__driver__frs-canary`

File: packages/test/test-service-load/package.json [hunk 4]
```diff
@@ -37,6 +37,7 @@
 		"prettier:fix": "prettier --write . --ignore-path ../../../.prettierignore",
 		"start": "node ./dist/nodeStressTest.js",
 		"start:frs": "node ./dist/nodeStressTest.js --driver routerlicious --driverEndpoint frs --profile ci_frs",
+		"start:frs:canary": "node ./dist/nodeStressTest.js --driver routerlicious --driverEndpoint frsCanary --profile ci_frs",
```
Comment (vladsud) [inline packages/test/test-service-load/package.json:40]: I really love your change overall, but I think this is wrong way to encode it (and we have same problem with ODSP).
The endpoint is FRS. What we deal here is ring. And there could be more rings in the future. For example, we might want to stand up "AFR in a box" dev-environment (as my team is getting to start making changes and might want to have early feedback here).

It's very useful to be able to see overall picture for a given service in telemetry, and it's somewhat weird that we have to do Data_driverEndPoint contains "frs" for that - I think we should have clean separation between type of service and rings.

Comment (sonalideshpandemsft) [reply packages/test/test-service-load/package.json:40]: By rings you mean frs, frs-canary, odsp, odsp-df and your service would be frs and odsp?

We can have a new option that differentiates the service and ring and I'm assuming the command would look something like: 
`"start:frs:canary": "node ./dist/nodeStressTest.js --driver routerlicious --service frs --ring canary --profile ci_frs",`

Comment (vladsud) [reply packages/test/test-service-load/package.json:40]: Yes, something like that.

Every service and every application have deployment rings. Canary / PROD are the ring for AFR, so is "SPDF / MSIT / PROD" for ODSP, or DF / PROD for Loop
Basically, same builds are showing up in rings at different times, but it's the same service, same endpoint (in a sense that API and behavior is exactly the same).

Comment (RishhiB) [reply packages/test/test-service-load/package.json:40]: service might be redundant with driverEndpoint

**Pull request merged**